src/setup_payload/python/GenerateSetupPayload.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python3
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+import sys
+import enum
+import argparse
+from bitarray import bitarray
+from bitarray.util import ba2int
+from stdnum.verhoeff import calc_check_digit
+import Base38
+
+# See section 5.1.4.1 Manual Pairing Code in the Matter specification v1.0
+kManualDiscriminatorLength = 4
+kPincodeLength = 27
+
+kManualChunk1DiscriminatorMsbitsLength = 2
+kManualChunk1DiscriminatorMsbitsPos = 0
+kManualChunk1VidPidPresentBitPos = kManualChunk1DiscriminatorMsbitsPos + kManualChunk1DiscriminatorMsbitsLength
+kManualChunk1Length = 1
+
+kManualChunk2DiscriminatorLsbitsLength = 2
+kManualChunk2PincodeLsbitsLength = 14
+kManualChunk2PincodeLsbitsPos = 0
+kManualChunk2DiscriminatorLsbitsPos = kManualChunk2PincodeLsbitsPos + kManualChunk2PincodeLsbitsLength
+kManualChunk2Length = 5
+
+kManualChunk3PincodeMsbitsLength = 13
+kManualChunk3PincodeMsbitsPos = 0
+kManualChunk3Length = 4
+
+kManualVendorIdLength = 5
+kManualProductIdLength = 5
+
+# See section 5.1.3. QR Code in the Matter specification v1.0
+kQRCodeVersionLength = 3
+kQRCodeDiscriminatorLength = 12
+kQRCodeVendorIdLength = 16
+kQRCodeProductIdLength = 16
+kQRCodeCommissioningFlowLength = 2
+kQRCodeDiscoveryCapBitmaskLength = 8
+kQRCodePaddingLength = 4
+kQRCodeVersion = 0
+kQRCodePadding = 0
+
+INVALID_PASSCODES = [00000000, 11111111, 22222222, 33333333, 44444444, 55555555,
+                     66666666, 77777777, 88888888, 99999999, 12345678, 87654321]
+
+
+class CommissioningFlow(enum.IntEnum):
+    Standard = 0,
+    UserIntent = 1,
+    Custom = 2
+
+
+class SetupPayload:
+    def __init__(self, discriminator, pincode, rendezvous=4, flow=CommissioningFlow.Standard, vid=0, pid=0):
+        self.longDiscriminator = discriminator
+        self.shortDiscriminator = self.GetShortDiscriminator(discriminator)
+        self.pincode = pincode
+        self.rendezvous = rendezvous
+        self.flow = flow
+        self.vid = vid
+        self.pid = pid
+
+    # 4 Most-significant bits of the 12-bits Discriminator
+    def GetShortDiscriminator(self, discriminator):
```

Comment:
Suggest returning the precomputed short discriminator here from self.short_discriminator

Reply:
This function computes the short discriminator. Constructor uses this function to precompute the short discriminator. So, we cannot return the precomputed value here.

I removed this method and moved the single line of code to constructor